mlrun/model_monitoring/api.py

```diff
@@ -242,6 +223,108 @@ def record_results(
+                                     `model_endpoints.spec.feature_stats`, a warning will be presented to the user.
+    :param drift_threshold:          The threshold of which to mark drifts. Should be similar to the drift threshold
+                                     that has already assigned to the current model endpoint.
+    :param possible_drift_threshold: The threshold of which to mark possible drifts. Should be similar to the possible
+                                     drift threshold  that has already assigned to the current model endpoint.
+
+    """
+    # Model path
+    if model_path and model_endpoint.spec.model_uri != model_path:
+        raise mlrun.errors.MLRunInvalidArgumentError(
+            f"provided model store path {model_path} does not match "
+            f"the path that is stored under the existing model "
+            f"endpoint record: {model_endpoint.spec.model_uri}"
+        )
+
+    # Feature stats
+    if (
+        sample_set_statistics
+        and sample_set_statistics != model_endpoint.status.feature_stats
+    ):
+        logger.warning(
+            "Provided sample set statistics is different from the registered statistics. "
+            "If you wish to use a new statistics as a reference expected data, it is "
+            "recommended to generate a new model endpoint record."
+        )
+    # drift and possible drift thresholds
+    if drift_threshold:
+        current_drift_threshold = model_endpoint.spec.monitor_configuration.get(
+            EventFieldType.DRIFT_DETECTED_THRESHOLD,
+            mlrun.mlconf.model_endpoint_monitoring.drift_thresholds.default.drift_detected,
+        )
+        if current_drift_threshold != drift_threshold:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Cannot change existing drift threshold. Expected {current_drift_threshold}, got {drift_threshold} "
+                f"Please update drift threshold or generate a new model endpoint record"
+            )
+
+    if possible_drift_threshold:
+        current_possible_drift_threshold = model_endpoint.spec.monitor_configuration.get(
+            EventFieldType.POSSIBLE_DRIFT_THRESHOLD,
+            mlrun.mlconf.model_endpoint_monitoring.drift_thresholds.default.possible_drift,
+        )
+        if current_possible_drift_threshold != possible_drift_threshold:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Cannot change existing possible drift threshold. Expected {current_possible_drift_threshold}, "
+                f"got {possible_drift_threshold}. Please update drift threshold or generate a new model endpoint record"
+            )
+
+
+def get_thresholds(
+    model_endpoint: ModelEndpoint,
+    drift_threshold: float = None,
+    possible_drift_threshold: float = None,
+) -> typing.Tuple[float, float]:
+    """
+    Generate drift and possible drift thresholds. If one of the thresholds is missing, will try to retrieve
```

Comment:
Using "Generate" is misleading.
Suggest calling the function retrieve_thresholds_from_model_endpoint() and in the docstring user retrieve

Reply:
I'll replace the `Generate` with `Get`. 
As for the function name, it's not correct that it retrieves it just from the model endpoint. If it's not exist, we retrieve it from mlrun config. Perhaps I'll rename it to `get_drift_thresholds_if_not_none`.